iothub/device/samples/PnpDeviceSamples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -101,21 +165,75 @@ public async Task PerformOperationsAsync(CancellationToken cancellationToken)
                 await SendTemperatureAsync(Thermostat1, cancellationToken);
                 await SendTemperatureAsync(Thermostat2, cancellationToken);
                 await SendDeviceMemoryAsync(cancellationToken);
+                await SendDeviceHealthTelemetryAsync(Thermostat1, cancellationToken);
 
                 temperatureReset = _temperature[Thermostat1] == 0 && _temperature[Thermostat2] == 0;
-                await Task.Delay(5 * 1000);
+                await Task.Delay(5 * 1000, CancellationToken.None);
             }
         }
 
+        private async Task SendInitialPropertyUpdatesAsync(string componentName, CancellationToken cancellationToken)
+        {
+            const string initialValueName = "initialValue";
+            var initialValue = new ThermostatInitialValue()
+            {
+                Temperature = 55,
+                Humidity = 68
+            };
+
+            PropertyCollection propertyPatch = PropertyConvention.CreatePropertyPatch(initialValueName, initialValue, componentName, s_propertyConvention);
+
+            await _deviceClient.UpdatePropertiesAsync(propertyPatch, cancellationToken);
+            _logger.LogDebug($"Property: Update - component=\"{componentName}\", {{\"{initialValueName}\" is complete.");
+        }
+
+        // Send the device health status over telemetry.
+        private async Task SendDeviceHealthTelemetryAsync(string componentName, CancellationToken cancellationToken)
+        {
+            string deviceHealthName = "deviceHealth";
+            var deviceHealth = new DeviceHealth
+            {
+                Status = "Running",
+                RunningTimeInSeconds = s_stopwatch.Elapsed.TotalSeconds,
+                IsStopRequested = false,
+                StartTime = s_applicationStartTime,
+            };
+
+            IDictionary<string, object> telemetryPayload = TelemetryConvention.FormatTelemetryPayload(deviceHealthName, deviceHealth);
+
+            using var message = new Message(telemetryPayload, s_telemetryConvention)
```

Comment:
What happened to the notion of an overload for Message ctor that takes an object that conforms to an interface to get the json value out of it?

Reply:
this is a more generic version of that, which takes in a dict of telemetry key-value pairs, so you can send multiple telemetry pairs within the same Message body. (we have a "helper" to assist in construction of that payload)